cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -144,24 +153,33 @@ public void init(ProgramController.State state) {
 
       @Override
       public void stopped() {
-        store.setStop(programId, runId,
-                      TimeUnit.SECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS),
-                      ProgramController.State.STOPPED);
-        LOG.debug("Program {} {} {} completed successfully.",
-                  programId.getNamespaceId(), programId.getApplicationId(), programId.getId());
-        latch.countDown();
+        try {
+          store.setStop(programId, runId,
+                        TimeUnit.SECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS),
+                        ProgramController.State.STOPPED);
+        } catch (NotFoundException e) {
+          throw Throwables.propagate(e);
+        } finally {
+          LOG.debug("Program {} {} {} completed successfully.",
+                    programId.getNamespaceId(), programId.getApplicationId(), programId.getId());
+          latch.countDown();
+        }
       }
 
       @Override
       public void error(Throwable cause) {
-        store.setStop(programId, runId,
-                      TimeUnit.SECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS),
-                      ProgramController.State.ERROR);
-        LOG.debug("Program {} {} {} execution failed.",
-                  programId.getNamespaceId(), programId.getApplicationId(), programId.getId(),
-                  cause);
-
-        latch.countDown();
+        try {
+          store.setStop(programId, runId,
+                        TimeUnit.SECONDS.convert(System.currentTimeMillis(), TimeUnit.MILLISECONDS),
+                        ProgramController.State.ERROR);
+        } catch (NotFoundException e) {
+          throw Throwables.propagate(e);
+        } finally {
+          LOG.debug("Program {} {} {} execution failed.",
```

Comment:
If you move the log message before `store.setStop` then you won't need the finally block.